java-checks/src/main/java/org/sonar/java/checks/UtilityClassWithPublicConstructorCheck.java

```diff
@@ -125,4 +139,16 @@ private static boolean hasPublicModifier(MethodTree methodTree) {
     return ModifiersUtils.hasModifier(methodTree.modifiers(), Modifier.PUBLIC);
   }
 
+  private static class SpringBootVisitor extends BaseTreeVisitor {
+
+    private static final MethodMatcher SPRING_BOOT = MethodMatcher.create().typeDefinition("org.springframework.boot.SpringApplication").name("run").withAnyParameters();
+    boolean springBootDetected;
+
+    @Override
+    public void visitMethodInvocation(MethodInvocationTree tree) {
+      if (SPRING_BOOT.matches(tree)) {
```

Comment:
I think it is better not to raise the issue, it is quite unlikely that such construct will be used in a utility class, which is the target for the rule. Let's assume user knows what he is doing. I added the testcase for completeness sake.